Helper/Data.php

```diff
@@ -849,13 +849,13 @@ public function isPaymentMethodOpenInvoiceMethod($paymentMethod)
     {
         if (strlen($paymentMethod) >= 9 && substr($paymentMethod, 0, 9) == 'afterpay_') {
             return true;
-        } else {
-            if ($paymentMethod == 'klarna' || $paymentMethod == 'ratepay') {
-                return true;
-            } else {
-                return false;
-            }
+        } elseif (strlen($paymentMethod) >= 6 && substr($paymentMethod, 0, 6) == 'klarna') {
```

Comment:
strlen can be omitted here since substr will return a string containing at most 6 characters
we could even switch to strpos($paymentMethod, 'klarna') === 0 to avoid the letter counting